airflow/models/baseoperator.py

```diff
@@ -124,7 +124,7 @@
 
 
 def parse_retries(retries: Any) -> int | None:
-    if retries is None or isinstance(retries, int):
+    if retries is None or type(retries) == int:
```

Comment:
Can you add a test for this

Reply:
@ephraimbuddy , I added tests and raised another PR - https://github.com/apache/airflow/pull/37183 . I couldn't pull the changes from this PR because it is failing a lint check and hence, i could not take commit , even with a cherry-pick, from this PR